File: scripts/check_ek_crm.js
const fs = require('fs');
const https = require('https');

const SUPABASE_URL = "https://gaawloviqgyzmqbtjsmd.supabase.co";
const SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdhYXdsb3ZpcWd5em1xYnRqc21kIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MzIwMDQwMiwiZXhwIjoyMDg4Nzc2NDAyfQ.bFDCYGlwGLfy50pxS1f0G4uyNOrZS3qBXcsG1wJSKqs";

function fetchSupabase(path) {
  return new Promise((resolve, reject) => {
    const url = new URL(`${SUPABASE_URL}/rest/v1/${path}`);
    const options = {
      headers: {
        'apikey': SUPABASE_KEY,
        'Authorization': `Bearer ${SUPABASE_KEY}`,
        'Content-Type': 'application/json'
      }
    };
    https.get(url, options, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          resolve(JSON.parse(data));
        } catch (e) {
          resolve({ error: e.message, raw: data });
        }
      });
    }).on('error', reject);
  });
}

async function main() {
  const logLines = [];
  const log = (msg) => {
    logLines.push(msg);
    console.log(msg);
  };

  log("=== 1. BUSCANDO PROYECTOS ===");
  const projects = await fetchSupabase("projects?select=id,name,client_id,status,project_analysis");
  if (Array.isArray(projects)) {
    projects.forEach(p => {
      log(`- Proj ID: ${p.id} | Name: ${p.name} | Status: ${p.status}`);
      if (p.project_analysis) {
        log(`  Transcripts count: ${(p.project_analysis.transcripts || []).length}`);
      }
    });
  } else {
    log(`Error fetching projects: ${JSON.stringify(projects)}`);
  }

  log("\n=== 2. BUSCANDO CLIENTES ===");
  const clients = await fetchSupabase("clients?select=id,name,email,company");
  if (Array.isArray(clients)) {
    clients.forEach(c => {
      log(`- Client ID: ${c.id} | Name: ${c.name} | Email: ${c.email} | Company: ${c.company}`);
    });
  }

  log("\n=== 3. BUSCANDO LEADS ===");
  const leads = await fetchSupabase("leads_cuentas?select=id,empresa,contacto_nombre,email,transcript_text");
  if (Array.isArray(leads)) {
    leads.forEach(l => {
      if ((l.email && l.email.toLowerCase().includes('leandro')) || (l.contacto_nombre && l.contacto_nombre.toLowerCase().includes('leandro')) || (l.empresa && l.empresa.toLowerCase().includes('ek'))) {
        log(`- Lead ID: ${l.id} | Empresa: ${l.empresa} | Contacto: ${l.contacto_nombre} | Email: ${l.email}`);
        log(`  Transcript length: ${l.transcript_text ? l.transcript_text.length : 0}`);
        if (l.transcript_text) {
          log(`  Transcript preview: ${l.transcript_text.slice(0, 200)}...`);
        }
      }
    });
  }

  fs.mkdirSync('.tmp', { recursive: true });
  fs.writeFileSync('.tmp/check_ek_crm_result.json', JSON.stringify({ projects, clients, leads }, null, 2));
  fs.writeFileSync('.tmp/check_ek_crm_log.txt', logLines.join('\n'));
}

main().catch(console.error);

File: scripts/check_ek_crm.py
import os
import json
import urllib.request
import urllib.parse

SUPABASE_URL = "https://gaawloviqgyzmqbtjsmd.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdhYXdsb3ZpcWd5em1xYnRqc21kIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MzIwMDQwMiwiZXhwIjoyMDg4Nzc2NDAyfQ.bFDCYGlwGLfy50pxS1f0G4uyNOrZS3qBXcsG1wJSKqs"

headers = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}",
    "Content-Type": "application/json"
}

output_lines = []

def log(msg):
    output_lines.append(str(msg))
    print(msg, flush=True)

def supabase_get(endpoint):
    url = f"{SUPABASE_URL}/rest/v1/{endpoint}"
    req = urllib.request.Request(url, headers=headers)
    try:
        with urllib.request.urlopen(req, timeout=10) as resp:
            return json.loads(resp.read().decode('utf-8'))
    except Exception as e:
        log(f"Error fetching {endpoint}: {e}")
        return []

log("=== BUSCANDO PROYECTO 'EK CRM' ===")
projects = supabase_get("projects?select=id,name,client_id,status,project_analysis")
matching_projects = [p for p in projects if "ek" in p.get("name", "").lower() or "crm" in p.get("name", "").lower()]
log(f"Proyectos encontrados ({len(matching_projects)}):")
for p in matching_projects:
    log(f"  - ID: {p['id']} | Nombre: {p['name']} | Status: {p.get('status')}")

log("\n=== BUSCANDO CLIENTES CON 'leandrogino@gmail.com' O 'EK' ===")
clients = supabase_get("clients?select=id,name,email,company")
matching_clients = [c for c in clients if "leandro" in (c.get("email") or "").lower() or "ek" in (c.get("name") or "").lower() or "ek" in (c.get("company") or "").lower()]
log(f"Clientes encontrados ({len(matching_clients)}):")
for c in matching_clients:
    log(f"  - ID: {c['id']} | Nombre: {c.get('name')} | Email: {c.get('email')} | Empresa: {c.get('company')}")

log("\n=== BUSCANDO LEADS CON 'leandrogino@gmail.com' ===")
leads = supabase_get("leads_cuentas?select=id,empresa,contacto_nombre,email,transcript_text")
matching_leads = [l for l in leads if "leandro" in (l.get("email") or "").lower() or "leandrogino" in (l.get("email") or "").lower()]
log(f"Leads encontrados ({len(matching_leads)}):")
for l in matching_leads:
    log(f"  - ID: {l['id']} | Empresa: {l.get('empresa')} | Nombre: {l.get('contacto_nombre')} | Email: {l.get('email')}")
    if l.get("transcript_text"):
        log(f"    Transcript presente: {len(l['transcript_text'])} caracteres")
    else:
        log(f"    Transcript: Vacio")

os.makedirs(".tmp", exist_ok=True)
with open(".tmp/check_ek_crm.txt", "w", encoding="utf-8") as f:
    f.write("\n".join(output_lines))
